core/store/src/trie/trie_storage.rs

```diff
@@ -16,7 +18,23 @@ pub struct TrieCache(Arc<Mutex<LruCache<CryptoHash, Arc<[u8]>>>>);
 
 impl TrieCache {
     pub fn new() -> Self {
-        Self(Arc::new(Mutex::new(LruCache::new(TRIE_MAX_CACHE_SIZE))))
+        Self::with_capacity(TRIE_MAX_SHARD_CACHE_SIZE)
+    }
+
+    pub fn with_capacity(cap: usize) -> Self {
+        Self(Arc::new(Mutex::new(LruCache::new(cap))))
+    }
+
+    pub fn get(&self, key: &CryptoHash) -> Option<Arc<[u8]>> {
+        self.0.lock().expect(POISONED_LOCK_ERR).get(key).cloned()
+    }
+
+    pub fn put(&self, key: CryptoHash, value: Arc<[u8]>) {
+        self.0.lock().expect(POISONED_LOCK_ERR).put(key, value);
```

Comment:
I'm already doing it in `retrieve_raw_bytes`, as this condition was there before:
```
...
if val.len() < TRIE_LIMIT_CACHED_VALUE_SIZE {
    self.put_to_cache(*hash, val.clone());
}
```

We don't have problems with previously existing values, because key is always a hash of a value. I put a comment about it inside `TrieCachingStorage`.

Reply:
sure you can have it there -- but I'd also add it here too..

(as you want to have 'clear' behaviors of your classes - and here one method allows large objects and the other one doesn't - which is confusing)

Reply:
Put the explanation to `retrieve_raw_bytes`, because there is no need in `TrieCache::put` anymore.